roles/edpm_network_config/tasks/main.yml

```diff
@@ -23,6 +23,27 @@
     - edpm_network_config_tool == 'nmstate'
     - not edpm_network_config_tool_nmstate_override|bool
 
+- name: Validate edpm_network_config_template against configuration tool
+  vars:
+    _tmpl: "{{ edpm_network_config_template | from_yaml }}"
+  block:
+    - name: Assert nmstate compatible edpm_network_config_template
+      when:
+        - edpm_network_config_tool == 'nmstate' or edpm_network_config_nmstate | bool
+      ansible.builtin.assert:
+        that:
+          - _tmpl.interfaces is defined
+        msg: |
+          nmstate is enabled, but edpm_network_config_template doesn't seem correct.
```

Comment:
"is missing the interfaces configuration" seems more helpful.

Reply:
Depends on the final test content - for now it only checks that key, but there are some others we can check, for instance related to DNS